### PR TITLE
Test rustc flag and feature flags

### PR DESCRIPTION
**Pull request by damienmg**
PR #151 broke some usage of those flags that make `cargo raze` build file not work any more.

This PR adds those 2 flag to a library so we notice those breakage. It allow the change for "rustc_flags" and fix the "features" argument.

As a consequence passing `["--arg value"]` to rustc_flags no longer works but that's fine (it's a reasonable breaking change). One should either do `["--arg", "value"]` or `["--arg=value"]`.

### CODE REVIEW

**Comment by mfarrugi** [comment]
Ah, didn't think of it as a breaking change. LGTM.

What's WIP about this?

**Comment by damienmg** [comment]
For now I fixed the args for one and sent a PR on cargo-raze to fix it (it's a breaking change but one that make sense).

The crate_features is still broken, I am trying to figure out why.

**Comment by damienmg** [comment]
Ok the fix is there now :)

File: rust/private/rustc.bzl [hunk 1]
```diff
@@ -208,7 +208,7 @@ def rustc_compile_action(
     args.add("--emit=dep-info,link")
     args.add("--color", "always")
     args.add("--target", toolchain.target_triple)
-    args.add_all(ctx.attr.crate_features, before_each = "--cfg", format_each = "feature=%s")
+    args.add_all(ctx.attr.crate_features, before_each = "--cfg", format_each = "feature=\"%s\"")
```
Comment (mfarrugi) [inline rust/private/rustc.bzl]: nit; a little easier to read as 'feature="%s"'

This quoting is necessary because features can have spaces in them?

**Comment by damienmg** [comment]
Side-note: this change does not introduce the incompatible change, it was introduced by #151. It just make it explicit that there is a breaking change.

The error I had was okish but could be better: `error: Unrecognized option: 'cap-lints allow'`

The only difficulty was that it was coming from cargo-raze generated BUILD files.

**Comment by acmcarther** [comment]
Yeah, I understand that the original issue came from https://github.com/bazelbuild/rules_rust/pull/151. I was mostly just wondering if this was also a good time to make the user-facing error easier to understand. It doesn't seem to me that there will be another obvious opportunity unless someone files a bug.

Ultimate decision is up to you since you're the one that had to debug the original issue, so you're most familiar with how hard that is to do with the current output.

**Comment by damienmg** [comment]
If I were setting the flag myself in the BUILD file the error would have been easy to understand, I had to figure things out because the error came from generated BUILD file. 

I don't think it is worth the added complexity to output another error message (that would be probably 2-3 lines of code)

**Comment by acmcarther** [comment]
Done!

**Comment by damienmg** [comment]
Thanks!

**Pull request merged**